specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json

```diff
@@ -0,0 +1,3432 @@
+        },
+        "encryptionSettingsCollection": {
+          "$ref": "#/definitions/EncryptionSettingsCollection",
+          "description": "Encryption settings collection used be Azure Disk Encryption, can contain multiple encryption settings per disk or snapshot."
+        },
+        "diskIOPSReadWrite": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The number of IOPS allowed for this disk; only settable for UltraSSD disks. One operation can transfer between 4k and 256k bytes."
+        },
+        "diskMBpsReadWrite": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The bandwidth allowed for this disk; only settable for UltraSSD disks. MBps means millions of bytes per second - MB here uses the ISO notation, of powers of 10."
+        },
+        "diskIOPSReadOnly": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The total number of IOPS that will be allowed across all VMs mounting the shared disk as ReadOnly. One operation can transfer between 4k and 256k bytes."
+        },
+        "diskMBpsReadOnly": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The total throughput (MBps) that will be allowed across all VMs mounting the shared disk as ReadOnly. MBps means millions of bytes per second - MB here uses the ISO notation, of powers of 10."
+        },
+        "maxShares": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The maximum number of VMs that can attach to the disk at the same time. Value greater than one indicates a disk that can be mounted on multiple VMs at the same time."
+        },
+        "encryption": {
+          "$ref": "#/definitions/Encryption",
+          "description": "Encryption property can be used to encrypt data at rest with customer managed keys or platform managed keys."
+        },
+        "networkAccessPolicy": {
+          "$ref": "#/definitions/NetworkAccessPolicy"
+        },
+        "diskAccessId": {
+          "type": "string",
+          "description": "ARM id of the DiskAccess resource for using private endpoints on disks."
+        },
+        "tier": {
+          "type": "string",
+          "description": "Performance tier of the disk (e.g, P4, S10) as described here: https://azure.microsoft.com/en-us/pricing/details/managed-disks/. Does not apply to Ultra disks."
+        },
+        "burstingEnabled": {
+          "type": "boolean",
+          "description": "Set to true to enable bursting beyond the provisioned performance target of the disk. Bursting is disabled by default. Does not apply to Ultra disks."
+        },
+        "purchasePlan": {
+          "$ref": "#/definitions/DiskPurchasePlan",
```

Comment:
We are changing this type from PurchasePlan to DiskPurchasePlan because they remove the PromotionCode in the original `PurchasePlan` model and we need it here?
I am asking this because this is causing a SDK breaking change across the api-versions

Reply:
the DiskRP backend still supports PromotionCode, so this will be needed. @vikramd-ms what should we do here?

Reply:
also, why is this causing breaking change? the definition did not change, only the name of the definition was changed here.

Reply:
Changing the type name changes method signatures and makes the code think the type changed, making it a breaking change.
Please revert this change.

Reply:
We will need to use DiskPurcahsePlan because it contains PromotionCode while PurchasePlan does not and DiskRP does return a PromotionCode property. So we are trying to *avoid* a breaking change here. @ArcturusZhang, please clarify how using DiskPurchasePlan is causing a breaking change and how we should proceed.

Reply:
The swagger definition of [ PurchasePlan has PromotionCode](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json#L2588), is that incorrect? 
Why is the name change needed?

Reply:
Oh, are you referring to the compute.json definition?
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json#L8340

Reply:
since we're not using that definition, it should be fine to use PurchasePlan itself right?

Reply:
Use the PurchasePlan in disk.json will trigger the conflicts with compute.json because different definition (with/without promotion code). DiskPurchasePlan is added to define the purchase plan on diskRP side. It is not a name for parameter, and it is a name for defination. We have done that for diskEncryptionSet before.

Reply:
I reverted the change back to PurchasePlan, and also ran the tests again, all the tests in sdk pass. The only check failing is the SDK track 2 validation, and since its not a required check this change should be fine. 
@ArcturusZhang , @Sandido. Please review this PR again. Have also attached the dot net sdk PR - https://github.com/Azure/azure-sdk-for-net/pull/19092